### PR TITLE
[Qol] Load i18n en locales during tests

### PR DESCRIPTION
**Pull request by flx-sta**
## Why am I making these changes?

Right now it's kinda hard to read the obfuscated logs during tests as it's just i18n keys.

## What are the changes from a developer perspective?

- Added a MSW that handles loading the locales files during tests
- Fix all tests that were written in a way that they were affected by this. They shouldn't anymore in the future now.

### Screenshots/Videos

N/A

## How to test the changes?

Run the tests. YOu should see human readable english output yet fully functional tests.

## Checklist
- [x] **I'm using `beta` as my base branch**
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- ~~[ ] Have I considered writing automated tests for the issue?~~
- ~~[ ] If I have text, did I make it translatable and add a key in the English locale file(s)?~~
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- ~~[] Are the changes visual?~~

### CODE REVIEW

File: src/test/mystery-encounter/encounters/teleporting-hijinks-encounter.test.ts [hunk 1]
```diff
@@ -1,21 +1,22 @@
+import BattleScene from "#app/battle-scene";
+import { TeleportingHijinksEncounter } from "#app/data/mystery-encounters/encounters/teleporting-hijinks-encounter";
 import * as MysteryEncounters from "#app/data/mystery-encounters/mystery-encounters";
+import { Abilities } from "#app/enums/abilities";
 import { Biome } from "#app/enums/biome";
 import { MysteryEncounterType } from "#app/enums/mystery-encounter-type";
 import { Species } from "#app/enums/species";
```
Comment (DayKev) [inline src/test/mystery-encounter/encounters/teleporting-hijinks-encounter.test.ts:7]:
```suggestion
import { Abilities } from "#enums/abilities";
import { Biome } from "#enums/biome";
import { MysteryEncounterType } from "#enums/mystery-encounter-type";
import { Species } from "#enums/species";
```

File: src/test/mystery-encounter/encounters/teleporting-hijinks-encounter.test.ts [hunk 2]
```diff
@@ -1,21 +1,22 @@
...
+import { CommandPhase } from "#app/phases/command-phase";
+import { MysteryEncounterPhase } from "#app/phases/mystery-encounter-phases";
+import { SelectModifierPhase } from "#app/phases/select-modifier-phase";
 import GameManager from "#app/test/utils/gameManager";
```
Comment (DayKev) [inline src/test/mystery-encounter/encounters/teleporting-hijinks-encounter.test.ts:11]:
```suggestion
import GameManager from "#test/utils/gameManager";
```

File: src/test/ui/type-hints.test.ts [hunk 3]
```diff
@@ -8,6 +8,7 @@ import GameManager from "#test/utils/gameManager";
 import Phaser from "phaser";
 import { afterEach, beforeAll, beforeEach, describe, expect, it } from "vitest";
 import MockText from "../utils/mocks/mocksContainer/mockText";
```
Comment (DayKev) [inline src/test/ui/type-hints.test.ts:10]:
```suggestion
import MockText from "#test/utils/mocks/mocksContainer/mockText";
```

**Review by DayKev — APPROVED** [review]

File: src/test/mystery-encounter/encounters/a-trainers-test-encounter.test.ts [hunk 4]
```diff
@@ -106,7 +107,8 @@ describe("A Trainer's Test - Mystery Encounter", () => {
       expect(scene.getCurrentPhase()?.constructor.name).toBe(CommandPhase.name);
       expect(enemyField.length).toBe(1);
       expect(scene.currentBattle.trainer).toBeDefined();
-      expect(["trainerNames:buck", "trainerNames:cheryl", "trainerNames:marley", "trainerNames:mira", "trainerNames:riley"].includes(scene.currentBattle.trainer!.config.name)).toBeTruthy();
+      console.log(scene.currentBattle.trainer?.config.name);
```
Comment (MokaStitcher) [inline src/test/mystery-encounter/encounters/a-trainers-test-encounter.test.ts:110]:
```suggestion
```

forgotten log?

**Review by MokaStitcher — APPROVED** [review]

**Comment by MokaStitcher** [comment]
don't know if this is just normal variance because the sample is small, but test runs with this seem to take longer on average than the other runs

seem to take ~3mins on average
Screenshot 2024-10-03 at 16 40 13
vs for other PRs it seems to be more around 2m15-2m30
Screenshot 2024-10-03 at 16 40 42

Not sure if much can be done about this?

**Comment by flx-sta** [comment]
> don't know if this is just normal variance because the sample is small, but test runs with this seem to take longer on average than the other runs
> 
> seem to take ~3mins on average (image) vs for other PRs it seems to be more around 2m15-2m30 (image)
> 
> Not sure if much can be done about this?

Not really. It's the nature of allowing requests to be sent and having to wait for them.
Here is the 3 states:
- before lazy load: All the locales were there ready with the build
- after lazy-load (now): i18next was mocked away completely so less to build and less to load
- this PR: i18next is allowed to send it's requests and they are redirected by MSW towards loading the file. This causes an `import()` which is an async "request". MSW is generally much faster than an actually request but there is just so much a machine can do

File: global.d.ts [hunk 5]
```diff
@@ -0,0 +1,14 @@
+import type { SetupServerApi } from "msw/node";
+
+export {};
+
+declare global {
+  /**
+   * Only used in testing.
+   * Can technically be undefined/null but for ease of use we are going to assume it is always defined.
+   * Used to looad i18n files exclusively.
```
Comment (torranx) [inline global.d.ts:9]:
```suggestion
   * Used to load i18n files exclusively.
```

File: src/test/system/game_data.test.ts [hunk 6]
```diff
@@ -11,13 +11,15 @@ import * as account from "../../account";
 
 const apiBase = import.meta.env.VITE_API_BASE_URL ?? "http://localhost:8001";
 
-export const server = setupServer();
+/** We ned a custom server. For some reasons I can't extend the listeners of {@linkcode global.i18nServer} with {@linkcode global.i18nServer.use} */
```
Comment (torranx) [inline src/test/system/game_data.test.ts:14]:
```suggestion
/** We need a custom server. For some reasons I can't extend the listeners of {@linkcode global.i18nServer} with {@linkcode global.i18nServer.use} */
```

**Review by torranx — COMMENTED** [review]
few typos, would love to see more description on the PR, otherwise looks good

**Review by DayKev — APPROVED** [review]

File: src/test/mystery-encounter/encounters/field-trip-encounter.test.ts [hunk 7]
```diff
@@ -198,12 +198,12 @@ describe("Field Trip - Mystery Encounter", () => {
...
-      expect(modifierSelectHandler.options[1].modifierTypeOption.type.name).toBe("modifierType:TempStatStageBoosterItem.x_speed");
-      expect(modifierSelectHandler.options[2].modifierTypeOption.type.name).toBe("modifierType:ModifierType.AddPokeballModifierType.name");
-      expect(i18next.t).toHaveBeenCalledWith("modifierType:ModifierType.AddPokeballModifierType.name", expect.objectContaining({ modifierCount: 5 }));
-      expect(modifierSelectHandler.options[3].modifierTypeOption.type.name).toBe("modifierType:ModifierType.IV_SCANNER.name");
-      expect(modifierSelectHandler.options[4].modifierTypeOption.type.name).toBe("modifierType:ModifierType.RARER_CANDY.name");
+      expect(modifierSelectHandler.options[0].modifierTypeOption.type.name).toBe(i18next.t("modifierType:TempStatStageBoosterItem.x_accuracy"));
+      expect(modifierSelectHandler.options[1].modifierTypeOption.type.name).toBe(i18next.t("modifierType:TempStatStageBoosterItem.x_speed"));
+      expect(modifierSelectHandler.options[2].modifierTypeOption.type.name).toBe(i18next.t("modifierType:ModifierType.AddPokeballModifierType.name", { modifierCount: 5, pokeballName: i18next.t("pokeball:greatBall")}));
```
Comment (DayKev) [inline src/test/mystery-encounter/encounters/field-trip-encounter.test.ts:203]:
```suggestion
      expect(modifierSelectHandler.options[2].modifierTypeOption.type.name).toBe(i18next.t("modifierType:ModifierType.AddPokeballModifierType.name", { modifierCount: 5, pokeballName: i18next.t("pokeball:greatBall") }));
```

**Review by DayKev — DISMISSED** [review]

**Review by torranx — DISMISSED** [review]

**Review by DayKev — APPROVED** [review]

**Review by torranx — APPROVED** [review]

**Review by MokaStitcher — APPROVED** [review]

**Pull request merged**